### PR TITLE
Add example of translated goto function

### PR DESCRIPTION
**Pull request by NlightNFotis**
Add an example C program and its equivalent goto-functions as produced by cbmc in the
goto-programs folder documentation.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [x] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/goto-programs/README.md [hunk 1]
```diff
@@ -524,6 +524,49 @@ then it performs 'linking' of the temporary into a passed destination
...
 [here](\ref section-linking-goto-models).
 
+\subsection subsection-goto-function-example Example goto-program
+
+The following C program, in a file named `mult.c`:
+
+```
+unsigned mult(unsigned a, unsigned b)
+{
+  int acc, i;
```
Comment (owen-mc-diffblue) [inline src/goto-programs/README.md]: Surely you should initialise `acc` to 0, to make this program make more sense?

**Comment by johnnonweiler** [comment]
I think this would fit in better together with the goto conversion section http://cprover.diffblue.com/group__goto-programs.html#goto-conversion .
@smowton deleted the placeholder for this example code, so he probably has opinions about this.

File: src/goto-programs/README.md [hunk 2]
```diff
@@ -524,6 +524,49 @@ then it performs 'linking' of the temporary into a passed destination
...
+     2: mult#return_value = (unsigned int)acc;
+        // 8 file third.c line 5 function mult
+        dead i;
+        // 9 file third.c line 5 function mult
+        dead acc;
+        // 10 file third.c line 6 function mult
+        END_FUNCTION
+```
+
+(The above result was produced using `cbmc version 5.11`)
```
Comment (tautschnig) [inline src/goto-programs/README.md:571]: Of course our documentation will always be up-to-date so that isn't needed :-)

**Review by tautschnig — APPROVED** [review]

**Comment by smowton** [comment]
@johnnonweiler @NlightNFotis I have no opinion on this

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 15c6b550f224dad29d9f52eac10d817b23335426).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/99762376

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 1ee96969a5c11ac179a2931afb03040a91a3c580).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/99784681

**Pull request merged**